plugins/woocommerce/src/Database/Migrations/CustomOrderTable/WPPostToCOTMigrator.php

```diff
@@ -0,0 +1,540 @@
+	 */
+	public function process_next_migration_batch( $batch_size = 100 ) {
+		global $wpdb;
+		$order_by = 'ID ASC';
+
+		$order_data = $this->order_table_migrator->fetch_data_for_migration( $this->get_where_clause(), $batch_size, $order_by );
+
+		foreach ( $order_data['errors'] as $post_id => $error ) {
+			$this->error_logger->log( 'info', "Error in importing post id $post_id: " . print_r( $error, true ) );
+		}
+
+		if ( count( $order_data['data'] ) === 0 ) {
+			return true;
+		}
+
+		$queries = $this->order_table_migrator->generate_insert_sql_for_batch( $order_data['data'], 'insert' );
+		$result  = $wpdb->query( $queries ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- $queries is already prepared.
+		if ( count( $order_data['data'] ) !== $result ) {
+			// Some rows were not inserted.
+			// TODO: Find and log the entity ids that were not inserted.
+			echo ' error ';
+		}
+
+		$order_post_ids = array_column( $order_data['data'], 'post_id' );
+		$this->process_next_migrator_batch( $this->billing_address_table_migrator, $order_post_ids, $order_by );
+		$this->process_next_migrator_batch( $this->shipping_address_table_migrator, $order_post_ids, $order_by );
+		$this->process_next_migrator_batch( $this->operation_data_table_migrator, $order_post_ids, $order_by );
+
+		$last_post_migrated = max( array_keys( $order_data['data'] ) );
+		$this->update_checkpoint( $last_post_migrated );
+
+		return false;
+	}
+
+	/**
+	 * Process next batch for a given address type.
+	 *
+	 * @param MetaToCustomTableMigrator $migrator Migrator instance for address type.
+	 * @param array                     $order_post_ids Array of post IDs for orders.
+	 * @param string                    $order_by Order by clause.
+	 */
+	private function process_next_migrator_batch( $migrator, $order_post_ids, $order_by ) {
+		global $wpdb;
+		$post_ids_where_clause = $this->get_where_id_clause( $order_post_ids, 'post_id' );
+		$batch_size            = count( $order_post_ids );
+		$data                  = $migrator->fetch_data_for_migration( $post_ids_where_clause, $batch_size, $order_by );
+		foreach ( $data['errors'] as $order_id => $error ) {
+			// TODO: Add name of the migrator in error message.
+			$this->error_logger->log( 'info', "Error in importing data for Order ID $order_id: " . print_r( $error, true ) );
+		}
+		$queries = $migrator->generate_insert_sql_for_batch( $data['data'], 'insert' );
+		$result  = $wpdb->query( $queries ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- Insert statements should already be escaped.
+		if ( count( $data['data'] ) !== $result ) {
+			// Some rows were not inserted.
+			// TODO: Find and log the entity ids that were not inserted.
+			echo 'error';
+		}
+	}
+
+	/**
+	 * Method to migrate single record.
+	 *
+	 * @param int $post_id Post ID of record to migrate.
+	 *
+	 * @return bool|\WP_Error
+	 */
+	public function process_single( $post_id ) {
```

Comment:
This might be intentional but thought I'd ask: shouldn't `process_single()` also migrate other meta (addresses and op data) for the migrated order?
I know this is not being used anywhere right now, but it's still a public function and might become handy later.

Reply:
Agreed, I was hoping to do it for later, but let me do it now.